libp2p/pubsub/pubsub.py

```diff
@@ -488,9 +504,20 @@ def handle_subscription(
 
         # TODO: Implement throttle on async validators
 
-        if len(async_topic_validator_futures) > 0:
-            results = await asyncio.gather(*async_topic_validator_futures)
-            if not all(results):
+        if len(async_topic_validators) > 0:
+            # TODO: Use a better pattern
```

Comment:
I think this is what you're looking for:

https://github.com/eabrouwer3/aiotoolz/blob/10790c9c5a8413502d8f35ce157966290492dbab/aiotoolz/functoolz.py#L536

I'd just vendor a copy of that function somewhere here in the codebase since that library looks dubiously maintained.

Reply:
Thanks for the direction. `pipe` looks nice to me for its clear syntax, but I also want to schedule all async validators at once and execute them *concurrently*. Does it seem that something like `gather` is quite suitable this case?